### PR TITLE
Fix compile issues

### PR DESCRIPTION
**Pull request by johnoloughlin**

### CODE REVIEW

File: cnivpp/localdb.go [hunk 1]
```diff
@@ -51,8 +52,9 @@ const debugVppDb = false
 // This structure is a union of all the VPP data (for all types of
 // interfaces) that need to be preserved for later use.
 type VppSavedData struct {
-	SwIfIndex     uint32 `json:"swIfIndex"`     // Software Index, used to access the created interface, needed to delete interface.
-	MemifSocketId uint32 `json:"memifSocketId"` // Memif SocketId, used to access the created memif Socket File, used for debug only.
+	l2SwIfIndex        interface_types.InterfaceIndex         `json:"swIfIndex"`
```
Comment (Eoghan1232) [inline cnivpp/localdb.go:55]: do we need both `l2SwIfIndex` and `interfaceSwIfIndex` ? if they are the same thing?

Comment (johnoloughlin) [reply cnivpp/localdb.go]: done

File: docker/userspacecni/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+RUN wget -qO- https://golang.org/dl/go1.20.1.linux-amd64.tar.gz | tar -C /usr/local -xz
+ENV PATH="${PATH}:/usr/local/go/bin"
+RUN go mod download
+RUN make generate
+RUN go mod tidy
+RUN make build
+#End of builder container
+
+# Copy build userspace cni bin to a small deployer container
+FROM ubuntu:22.04
```
Comment (Eoghan1232) [inline docker/userspacecni/Dockerfile:15]: This base image would be quite big?

Comment (johnoloughlin) [reply docker/userspacecni/Dockerfile:15]: I can change to a smaller image later when I get into a state where I can test all of this running.

File: cnivpp/api/memif/memif.go [hunk 3]
```diff
@@ -62,15 +64,16 @@ var stateStr = [...]string{"dn", "up"}
...
 // Attempt to create a MemIf Interface.
 // Input:
-//   ch api.Channel
-//   socketId uint32
-//   role MemifRole - RoleMaster or RoleSlave
-func CreateMemifInterface(ch api.Channel, socketId uint32, role MemifRole, mode MemifMode) (swIfIndex uint32, err error) {
+//
+//	ch api.Channel
+//	socketId uint32
+//	role MemifRole - RoleMaster or RoleSlave
```
Comment (walshric) [inline cnivpp/api/memif/memif.go:70]: RoleWorker

File: cnivpp/api/memif/memif.go [hunk 4]
```diff
@@ -169,19 +172,20 @@ func DumpMemif(ch api.Channel) {
...
 			modeStr[reply.Mode],
 			string(reply.IfName),
-			macAddr.String(),
+			macAddr,
 			reply.RingSize,
-			reply.BufferSize,
-			stateStr[reply.AdminUpDown],
-			stateStr[reply.LinkUpDown])
+		)
+		// reply.BufferSize,
```
Comment (walshric) [inline cnivpp/api/memif/memif.go:186]: Should we just remove ? - or add comment why disabled

File: .github/workflows/static-scan.yml [hunk 5]
```diff
@@ -28,22 +28,31 @@ jobs:
     - uses: brpaz/hadolint-action@v1.2.1
       name: Run Hadolint
       with:
-       dockerfile: Dockerfile
+       dockerfile: ./docker/userspacecni/Dockerfile
```
Comment (walshric) [inline .github/workflows/static-scan.yml:31]: may want to consider a non relative path on a future push

File: cnivpp/api/bridge/bridge.go [hunk 6]
```diff
@@ -49,14 +48,14 @@ func CreateBridge(ch api.Channel, bridgeDomain uint32) error {
...
-		UuFlood: 1,
-		Forward: 1,
-		Learn:   1,
-		ArpTerm: 0,
+		Flood:   true,
+		UuFlood: true,
+		Forward: true,
+		Learn:   true,
+		ArpTerm: false,
 		MacAge:  0,
```
Comment (walshric) [inline cnivpp/api/bridge/bridge.go:56]: Why isnt MacAge false

**Review by walshric — APPROVED** [review]

File: cnivpp/api/bridge/bridge.go (hunk 6 above)
Comment (Eoghan1232) [reply cnivpp/api/bridge/bridge.go:56]: it's not a boolean, and is implemented as an int.

File: .github/workflows/static-scan.yml (hunk 5 above)
Comment (Eoghan1232) [reply .github/workflows/static-scan.yml:31]: This PR is a WIP for now, so this will change later

File: cnivpp/api/bridge/bridge.go (hunk 6 above)
Comment (walshric) [reply cnivpp/api/bridge/bridge.go:56]: @Eoghan1232  - So this is a mac aging time ? - as opposed to mac aging on/off ?

Comment (walshric) [reply cnivpp/api/bridge/bridge.go:56]: aging time in secs

File: go.mod [hunk 7]
```diff
@@ -1,23 +1,65 @@
 module github.com/intel/userspace-cni-network-plugin
 
-go 1.14
+go 1.19
```
Comment (Eoghan1232) [inline go.mod:3]: 1.20 ?

Actions stated 1.20, perhaps we should upgrade to it anyways :)

**Comment by johnoloughlin** [comment]
Needed to force push as I rebased on master

**Comment by johnoloughlin** [comment]
unsure where github.com/konsorten/go-windows-terminal-sequences v1.0.3/go.mod h1:T0+1ngSBFLxvqU3pZ+m/2kptfBszLMUkC4ZK/EgS/cQ= comes from. Any tests I do on my linux system doesn't get this package

**Review by Eoghan1232 — APPROVED** [review]

**Pull request merged**